faster_whisper/transcribe.py

```diff
@@ -623,6 +636,10 @@ def generate_with_fallback(
         max_initial_timestamp_index = int(
             round(options.max_initial_timestamp / self.time_precision)
         )
+        if options.max_new_tokens is not None:
+            max_length = min(self.max_length, len(prompt) + options.max_new_tokens)
```

Comment:
Can you re-check this logic ? I think we should use max_length option as input instead of creating new param max_new_tokens.
Example: `max_length = min(self.max_length, options.max_length) if options.max_new_tokens is not None else self.max_length`

Reply:
We can event simplify like:  
`max_length = options.max_length if options.max_length is not None else self.max_length`
  
And please rebase your branch on the last version of master to avoid conflict.

Reply:
thx for your advice, I use a new max_new_tokens is just for keeping the same logic as distil-whisper, where the setting is 
```python
pipe = pipeline(
    "automatic-speech-recognition",
    model=model,
    tokenizer=processor.tokenizer,
    feature_extractor=processor.feature_extractor,
    max_new_tokens=128,
    torch_dtype=torch_dtype,
    device=device,
)
```
and the default max_length seems include the length of the prompt. So, to make these two compatable, I introduce a new param, not sure if this is a good way, glad to hear your advice. thx a lot.

Reply:
I remove the new param max_new_tokens, and use the options.max_length instead, thx

Reply:
@metame-none, hello.
I think we should rollback to both use max_new_tokens and max_length params, then update this logic similar to what HF transformer is doing:
```python
# If the user passes `max_new_tokens`, increase its number to account for the prompt
if kwargs.get("max_new_tokens", None) is not None:
    kwargs["max_new_tokens"] += len(text_prompt_ids)
    if kwargs["max_new_tokens"] >= self.config.max_target_positions:
        raise ValueError(
            f"The length of the sliced `prompt_ids` is {len(text_prompt_ids)}, and the `max_new_tokens` "
            f"{kwargs['max_new_tokens'] - len(text_prompt_ids)}. Thus, the combined length of the sliced "
            f"`prompt_ids` and `max_new_tokens` is: {kwargs['max_new_tokens']}. This exceeds the "
            f"`max_target_positions` of the Whisper model: {self.config.max_target_positions}. "
            "You should either reduce the length of your prompt, or reduce the value of `max_new_tokens`, "
            f"so that their combined length is less that {self.config.max_target_positions}."
        )
```
=> raise error if max_new_tokens + len(prompt) > max_length
For more info, please check this [pr](https://github.com/huggingface/transformers/pull/26164).

Besides, I also agree that max_length includes the length of the prompt: https://huggingface.co/docs/transformers/main/en/main_classes/text_generation#transformers.GenerationConfig.max_length

Please take a look. Tks.